pkg/kubectl/cmd/patch.go

```diff
@@ -67,7 +67,7 @@ func NewCmdPatch(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 			cmdutil.CheckErr(err)
 		},
 	}
-	cmd.Flags().StringP("patch", "p", "", "The patch to be applied to the resource JSON file.")
+	cmd.Flags().StringP("patch", "p", "", "The patch to be applied to the resource JSON file, or - to use files on STDIN as the patch source.")
```

Comment:
This is kind of weird. JSON or `-`? I'd prefer another flag: --patch-file or somesuch.